apps/dashboard/pages/api/v1/github/verify.ts

```diff
@@ -0,0 +1,196 @@
+  }
+  if (!keyId) {
+    throw new Error("No KeyID found");
+  }
+  if (data?.length === 0) {
+    throw new Error("No data found");
+  }
+  // Worked but needs to be off while other functions are tested. Hashed with example from site so will always fail till live
+  const isGithubVerified = await verify_git_signature(rawBody.toString(), signature, keyId);
+  if (!isGithubVerified) {
+    throw new Error("Github signature not verified");
+  }
+
+  for (const item of data) {
+    const token = item.token.toString();
+    const hashedToken = await sha256(token);
+    const isKeysFound = await db.query.keys.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(eq(table.hash, hashedToken), isNull(table.deletedAt)),
+    });
+    const ws = await db.query.workspaces.findFirst({
+      where: (table, { and, eq, isNull }) =>
+        and(
+          eq(table.id, isKeysFound?.forWorkspaceId ? isKeysFound?.forWorkspaceId : ""),
+          isNull(table.deletedAt),
+        ),
+    });
+    if (!ws) {
+      throw new Error("workspace does not exist");
+    }
+    const users = await getUsers(ws.tenantId);
+    const date = new Date().toDateString();
+    for await (const user of users) {
+      await resend.sendLeakedKeyEmail({
+        email: user.email,
+        date: date,
+        source: item.source,
+        type: item.type,
+        url: item.url,
+      });
+      const text1 = `Type: ${item.type} \n Source: ${item.source} \n Date: ${date} \n URL: ${item.url}`;
+      const text2 = `Key: ${isKeysFound?.id} \n Workspace: ${ws.name} \n Tenant: ${ws.tenantId} \n User: ${user.email}`;
+      if (isKeysFound) {
+        alertSlack("Leaked Key Found", text1, text2);
+      }
+    }
+  }
+
+  return response.status(201).json({});
+}
+
+async function alertSlack(title: string, text1: string, text2: string): Promise<void> {
+  const url = process.env.SLACK_WEBHOOK_URL_LEAKED_KEY;
+  if (!url) {
+    throw new Error("Missing required environment variables");
+  }
+  await fetch(url, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      text: title,
+      blocks: [
+        {
+          type: "section",
+          fields: [
+            {
+              type: "mrkdwn",
+              text: text1,
+            },
+            {
+              type: "mrkdwn",
+              text: text2,
+            },
+          ],
+        },
+      ],
+    }),
+  }).catch((err: Error) => {
+    console.error(err);
+  });
+}
+
+async function getUsers(tenantId: string): Promise<{ id: string; email: string; name: string }[]> {
+  const userIds: string[] = [];
+  if (tenantId.startsWith("org_")) {
+    const members = await clerkClient.organizations.getOrganizationMembershipList({
+      organizationId: tenantId,
+    });
+    for (const m of members) {
+      userIds.push(m.publicUserData!.userId);
+    }
+  } else {
+    userIds.push(tenantId);
+  }
+
+  return await Promise.all(
```

Comment:
Potentially we want to use allSettled versus `all`

Then we can check which promise failed and then do something with it. 

Right now with a promise all, it is all or nothing.